### ISSUE TITLE
Fix semi-autgenerated documentation of processing algorithms

### ISSUE THREAD

**Issue by jakimowb** [EnMAP-Box/enmap-box-documentation#80]
As discussed on 07aug24

 # Problem
The semiautomatic generation of the processing algorithms documentation is cumbersome. As now this involves the following steps:

1. Manual generated rst code is located in (EDR) `source/usr_section/usr_manual/processing_algorithms\**.rst` files. 
2. EnMAP-Box source repo (ER) `scripts/create_processing_rst.py` generates rst files for (EDR) `source/usr_section/usr_manual/processing_algorithms/**.rst` files and the (EDR) `source/usr_section/usr_manual/processing_algorithms/processing_algorithms.rst`
3. If existing, the manually generated files (below '') are included below the Headline and first description. E.g. in  (EDR) `source/usr_section/usr_manual/processing_algorithms/auxilliary/build_3d_cube.rst`:

```
 .. _Build 3D Cube:

*************
Build 3D Cube
*************

Build an 3D Cube visualization of a (spectral) `raster layer <https://enmap-box.readthedocs.io/en/latest/general/glossary.html#term-raster-layer>`_, consisting of two individually stylable cube face and cube side `layers <https://enmap-box.readthedocs.io/en/latest/general/glossary.html#term-layer>`_. 

.. include:: ../../processing_algorithms_includes/auxilliary/build_3d_cube.rst

**Parameters**
```

This has the following disadvantages:

* manually defined rst code can only be added below the description and the `**Parameters**` line.
* There is no other way to _enhance_ the autgenerated rst documentation in other places, e.g. to add additional lines or screenshots to the _**Parameters**_ or _**Command-line usage**_ section.
* there is somehow a redundant structure. `source/usr_section/usr_manual/processing_algorithms` contains what we expect to see on the webpage, while `source/usr_section/usr_manual/processing_algorithms_includes` contains kind of a mirrored file structure as well. It is hard to find files that belong together (would be better to have auto- and manual generated files which belong together in the same folder)

Furthermore a couple of errors messages are shown, which are _far from explaining what the actual problem is_ :
```
 source\usr_section/usr_manual/processing_algorithms_includes/classification/class_fraction_layer_from_categorized_vector_layer.rst: WARNING: document isn't included in any toctree
...
```
The problem is that a `source/usr_section/usr_manual/processing_algorithms/classification/class_fraction_layer_from_categorized_vector_layer.rst` either does not exists, or exists but does not include the (existing) `source\usr_section/usr_manual/processing_algorithms_includes/classification/class_fraction_layer_from_categorized_vector_layer.rst`

# Proposed solution

Improve the (ER) `scripts/create_processing_rst.py` script to 
* take care on such issues and make the developer who syncs the docs with the processing algorithms aware of these inconsistencies
* simplify the way how auto-generated files are created, updated and manually edited, by
* avoid the separation between auto-generated and manually (and included) files. Instead, update sections but keep additional, manually added content.

A auto-generated / auto-updated rst file could look like
````
.. ## AUTOGENERATED TITLE START ##
.. _Build 3D Cube:

*************
Build 3D Cube
*************
.. ## AUTOGENERATED TITLE END##

Here I can add my manual defined rst code

.. ## AUTOGENERATED DESCRIPTION START ##
Build an 3D Cube visualization of a (spectral) `raster layer <https://enmap-box.readthedocs.io/en/latest/general/glossary.html#term-raster-layer>`_, consisting of two individually stylable cube face and cube side `layers <https://enmap-box.readthedocs.io/en/latest/general/glossary.html#term-layer>`_. 
.. ## AUTOGENERATED DESCRIPTION START ##

Here I can add more manual defined rst code

**Parameters**

And here too.. 

.. ## AUTOGENERATED PARAMETERS START ##
:guilabel:`Raster layer with features` [raster]
    The `raster layer <https://enmap-box.readthedocs.io/en/latest/general/glossary.html#term-raster-layer>`_ to be visualized.

````

`.. ## AUTOGENERATED` blocks will be updated (or added, if not existing), while line in between should be kept as is.

**Comment by janzandr** [comment]
> source\usr_section/usr_manual/processing_algorithms_includes/classification/class_fraction_layer_from_categorized_vector_layer.rst: WARNING: document isn't included in any toctree

Those warnings are (most likely) a result of typos in the RST filenames.

**Comment by jakimowb** [comment]
> > source\usr_section/usr_manual/processing_algorithms_includes/classification/class_fraction_layer_from_categorized_vector_layer.rst: WARNING: document isn't included in any toctree
> 
> Those warnings are (most likely) a result of typos in the RST filenames.

Nope. This warning comes if an *.rst is not reachible, e.g. through TOC entriers. User's can not navigate to.
Has been fixed by moving the code from `processing_algorithms_includes/**.rst` files into `processing_algorithms/**.rst` files (#87).
